File: src/cpp/FindNumbersWithSum.cpp
class Solution {
public:
    vector<int> FindNumbersWithSum(vector<int> array,int sum) {
      if (array.size() < 1) {
        return vector<int> ();
      }

      vector<int> result;

      int start = 0;
      int end = array.size() - 1;

      while (start < end) {
        int currSum = array[start] + array[end];

        if (currSum == sum) {
          result.push_back(array[start]);
          result.push_back(array[end]);
          break;
        }
        else if (currSum < sum) {
          start++;
        }
        else {
          end--;
        }
      }

      return result;
    }
};

File: src/java/FindNumbersWithSum.java
import java.util.ArrayList;
public class Solution {
	public ArrayList<Integer> FindNumbersWithSum(int [] array,int sum) {
        ArrayList<Integer> result = new ArrayList<>();

        if (array == null || array.length == 0) {
            return result;
        }

        int left = 0;
        int right = array.length - 1;

        while (left < right) {
            int temp = array[left] + array[right];
            if (temp == sum) {
                result.add(array[left]);
                result.add(array[right]);
                return result;
            } else if (temp > sum) {
                right--;
            } else {
                left++;
            }
        }

        return result;
    }
}
